lib/memoist.rb

```diff
@@ -179,8 +175,8 @@ def memoize(*method_names)
           # end
 
           module_eval <<-EOS, __FILE__, __LINE__ + 1
-            def #{method_name}(reload = false)
-              skip_cache = reload || !instance_variable_defined?("#{memoized_ivar}")
+            def #{method_name}(reload = false, memoist_reload: false)
+              skip_cache = reload || memoist_reload || !instance_variable_defined?("#{memoized_ivar}")
```

Comment:
@prashantvithani This is an standardization. Since memoist_reload is the only foolproof way of detecting a reload scenario when arity is not zero, it make sense to support it even when arity is zero. This won't break any existing use cases, since, in such cases the positional arg `reload` is still kept as it is.